library-best-practices/JLBP-2.md

```diff
@@ -2,19 +2,40 @@
 -----------------------------
 
 - Avoid exposing types from your dependencies.
-  - Your dependencies have their own rules about how often
-    they do major version bumps and make breaking changes.
-    Their frequency of such changes may be more often than yours,
-    which causes problems if you want to retain a stable surface for
-    longer than your dependencies. Historically this was a
-    big problem for Guava, which made a major version bump
-    every 6 months, although Guava no longer does this.
-- Use package-protected classes and methods when the API is internal
-  only and should not be used by consumers.
-- Do not mark methods and classes public by default.
-  Assume non-public until a need is known.
+
+  - Method arguments and return values from public methods should be standard Java 
+    types such as `java.time.LocalDate` or classes defined in the library itself,
+    not third party types such as `org.joda.time.LocalDate`.
```

Comment:
This is nice enhancement, first explaining the meaning of "exposing types from your dependencies"